tests/fracs/test_fracture_network_2d.py

```diff
@@ -0,0 +1,337 @@
+        network_2 = pp.create_fracture_network(fracs2)
+
+        # Add networks, check tags
+        together = network_1.add_network(network_2)
+        assert np.all(together._edges[2, 2:4] == tag2)
+
+        # Add networks reverse order
+        together = network_2.add_network(network_1)
+        assert np.all(together._edges[2, :2] == tag2)
+
+        # Assign tags to network1
+        tag1 = 2
+        # Had to change this to ensure that edges still has ``dtype np.int8``
+        network_1._edges = np.vstack(
+            (network_1._edges, tag1 * np.ones(2, dtype=np.int8))
+        )
+        together = network_1.add_network(network_2)
+        known_tags = np.array([tag1, tag1, tag2, tag2])
+        assert np.all(together._edges[2] == known_tags)
+
+    # Below are tests with polytopal domains
+
+    def test_triangle_domain_no_fracs(self):
+        # test a grid where the outer domain is a triangle, no fractures
+        line_1 = np.array([[0, 1], [0, 0]])
+        line_2 = np.array([[1, 0], [0, 1]])
+        line_3 = np.array([[0, 0], [1, 0]])
+        lines = [line_1, line_2, line_3]
+        domain = pp.Domain(polytope=lines)
+
+        network = pp.create_fracture_network(domain=domain)
+        self.check_mdg_from_polytopal_2d_domain(domain, network, [0, 0, 1])
+
+    def test_triangle_domain_with_fracs(self):
+        # test a grid where the outer domain is a triangle, one fracture immersed in the
+        # domain
+        line_1 = np.array([[0, 1], [0, 0]])
+        line_2 = np.array([[1, 0], [0, 1]])
+        line_3 = np.array([[0, 0], [1, 0]])
+        lines = [line_1, line_2, line_3]
+        domain = pp.Domain(polytope=lines)
+
+        frac = pp.LineFracture(np.array([[0, 1], [0, 1]]))
+        network = pp.create_fracture_network([frac], domain=domain)
+
+        self.check_mdg_from_polytopal_2d_domain(domain, network, [0, 1, 1])
+
+    def test_pentagon_domain_with_fracs(self):
+        # test a grid where the outer domain is a concave pentagon, one fracture is
+        # partially immersed which results in two 1d grids
+        line_1 = np.array([[0, 1], [0, 0]])
+        line_2 = np.array([[1, 1], [0, 1]])
+        line_3 = np.array([[1, 0.5], [1, 0.5]])
+        line_4 = np.array([[0.5, 0], [0.5, 1]])
+        line_5 = np.array([[0, 0], [1, 0]])
+        lines = [line_1, line_2, line_3, line_4, line_5]
+        domain = pp.Domain(polytope=lines)
+
+        frac = pp.LineFracture(np.array([[0.2, 0.8], [0.6, 0.6]]))
+        network = pp.create_fracture_network([frac], domain=domain)
+
+        self.check_mdg_from_polytopal_2d_domain(domain, network, [0, 2, 1])
+
+    # Test of other methods
+
+    def test_copy(self):
+        network_1 = pp.create_fracture_network(self.fracs)
+
+        copy = network_1.copy()
+        num_p = self.pts.shape[1]
+
+        network_1._pts = np.random.rand(2, num_p)
+        assert np.allclose(copy._pts, self.pts)
+
+    def test_no_snapping(self):
```

Comment:
Can we parametrize snapping methods? I think so, but I'm not sure.

Reply:
Changing it